### ISSUE TITLE
[7.0] web_pytz: prevents from connecting to instance

### ISSUE THREAD

**Issue by gurneyalex** [OCA/web#141]
When this addon is installed, the instance becomes unusable. 

```
TypeError: moment.tz is undefined
tz = moment.tz.zone(args.tz_name.toJSON()),
```

this is from web_pytz.js (line 55, col 12)

**Comment by gurneyalex** [comment]
@hbrunn if you have any idea what is amiss your advice is welcome.

**Comment by hbrunn** [comment]
I'd start with verifying that `static/lib/moment-timezone.min.js` is loaded

**Comment by OSguard** [comment]
i just figure out that this is the reason why runbot not working:
http://2947807-7-0-5b1147.runbot2.odoo-community.org/?db=2947807-7-0-5b1147-all&debug=

**Comment by OSguard** [comment]
after installing web_list_html_widget together with web_pytz it fails, i will validate this with a new database

**Comment by OSguard** [comment]
now the error hits after installing "web_nocreatedb" Oo

**Comment by OSguard** [comment]
you need to install this 13 modules to reproduce the error:
            name             |                                 author  
-----------------------------+------------------------------------------------------------------------
 web_m2x_options             | Tuxservices,Odoo Community Association (OCA)
 web_list_html_widget        | initOS GmbH & Co. KG,Odoo Community Association (OCA)
 web_recipients_uncheck      | initOS GmbH & Co. KG,Odoo Community Association (OCA)
 web_search_alphabetic       | Serpent Consulting Services Pvt. Ltd.,Odoo Community Association (OCA)
 web_pytz                    | Therp BV,Odoo Community Association (OCA)
 web_popup_large             | Therp BV,Odoo Community Association (OCA)
 web_tree_many2one_clickable | Therp BV,Odoo Community Association (OCA)
 web_polymorphic_many2one    | Elico Corp,Odoo Community Association (OCA)
 web_send_message_popup      | Camptocamp,Odoo Community Association (OCA)
 web_relativedelta           | Therp BV,Odoo Community Association (OCA)
 web_widget_float_formula    | GRAP,Odoo Community Association (OCA)
 web_warning_sound           | HacBee UAB,Odoo Community Association (OCA)
 web_nocreatedb              | Vauxoo,Odoo Community Association (OCA)

**Comment by hbrunn** [comment]
@OSguard any chance `web_relativedelta` is the culprit? This also pulls `moment.js`, but in an older version (2.6.0 vs. 2.8.3). The first who loads wins, and it seems like that's the case for the 2.6.0 version. Maybe `moment-timezone.js` refuses to load with this version?

It _should_ not be a problem to simply update `web_relativedelta`'s `moment.js` For a quick and dirty test, I'd simply copy `web_pytz`'s `moment.js` to `web_relativedelta/static/lib`, reload and see what happens.

It might be even worth a try to update both modules to current upstream from http://momentjs.com

**Comment by hbrunn** [comment]
...if you do the update, don't repeat my mistake and use the non-minified versions. OpenERP minifies them anyways, and this way, you save a lot of headaches when debugging

**Comment by OSguard** [comment]
only installing both modules are fine. Having all other modules install always failing. So this was also my first idea but i find no evidence for that. But i can try to replace it in my broken installation. (but not today)

**Comment by hbrunn** [comment]
have a look at load order. On the runbot you linked, `web_pytz`'s momentjs is loaded first and then probably overwritten by `web_relativedelta`'s version. If on your isolated database this order is reversed, then things should be fine. So the trick is for your test to manipulate load order to test both cases.

**Comment by hbrunn** [comment]
Of course not today, now you code sprinters should go for the well deserved belgian beers!

**Comment by hbrunn** [comment]
another hypothesis: The `lib`-mechanism should avoid loading the same library multiple times, and to my knowledge, this goes by filename. But here, one is called `moment.min.js` while the other is `moment.js`, possibly this screws up things? Try changing the `min`-version to just `moment.js` and see if odoo still loads both. If everything works then, you might still give the update a try in order to avoid future version incompatibilities.

**Comment by OSguard** [comment]
- rename to `moment.js` loads both with the same name and error still there
- replacing the older version with '2.8.3' also did not solve the problem.

**Comment by OSguard** [comment]
removing one `moment.js` from one of the modules solves the problem.
I suggest either:
- merge the two modules in one
- make a single module with only holds the lib for moments.js

@hbrunn and @gurneyalex what do you think?

**Comment by hbrunn** [comment]
depending both on a module `web_moment` seems the proper choice then in my opinion.

**Comment by OSguard** [comment]
@hbrunn can you do this, as you are more familiar with this modules?

**Comment by hbrunn** [comment]
yes, but not before end of this week

**Comment by hbrunn** [comment]
@OSguard @gurneyalex I proposed a fix in #142

**Comment by hbrunn** [comment]
closed as of #142
